expression/expression.go

```diff
@@ -68,11 +68,21 @@ type VecExpr interface {
 	VecEvalJSON(ctx sessionctx.Context, input *chunk.Chunk, result *chunk.Column) error
 }
 
+// ReverseExpr contains all resersed evaluation methods.
+type ReverseExpr interface {
+	// SupportReverseEval checks whether the builtinFunc support reverse evaluation.
+	SupportReverseEval() bool
+
+	// ReverseEval evaluates the only one column value with given function result.
+	ReverseEval(sc *stmtctx.StatementContext, res types.Datum, rType types.RoundingType) (val types.Datum, err error)
```

Comment:
How about returning `(lowBound, upBound types.Datum, error)`?
Then we can remove the `RoundingType`.

Reply:
I prefer to keep the `RoundingType`. Because for index join, the range `[a, b)`, we only need the `ReverseEval(a, Ceiling)` and `ReverseEval(a, Floor)`. By the way, in the future, if we need a reverse calculation for both bound. We add one function by calling the `ReverseEval` twice with different `RoundingType`s to realize it.